idaes/models/unit_models/solid_liquid/tests/test_sl_separator.py

```diff
@@ -0,0 +1,492 @@
+
+    @pytest.mark.build
+    @pytest.mark.unit
+    def test_build(self, model):
+
+        assert isinstance(model.fs.unit.solid_inlet, Port)
+        assert len(model.fs.unit.solid_inlet.vars) == 4
+        assert hasattr(model.fs.unit.solid_inlet, "flow_vol")
+        assert hasattr(model.fs.unit.solid_inlet, "conc_mol_comp")
+        assert hasattr(model.fs.unit.solid_inlet, "temperature")
+        assert hasattr(model.fs.unit.solid_inlet, "pressure")
+
+        assert isinstance(model.fs.unit.solid_outlet, Port)
+        assert len(model.fs.unit.solid_outlet.vars) == 4
+        assert hasattr(model.fs.unit.solid_outlet, "flow_vol")
+        assert hasattr(model.fs.unit.solid_outlet, "conc_mol_comp")
+        assert hasattr(model.fs.unit.solid_outlet, "temperature")
+        assert hasattr(model.fs.unit.solid_outlet, "pressure")
+
+        assert isinstance(model.fs.unit.liquid_inlet, Port)
+        assert len(model.fs.unit.liquid_inlet.vars) == 4
+        assert hasattr(model.fs.unit.liquid_inlet, "flow_vol")
+        assert hasattr(model.fs.unit.liquid_inlet, "conc_mol_comp")
+        assert hasattr(model.fs.unit.liquid_inlet, "temperature")
+        assert hasattr(model.fs.unit.liquid_inlet, "pressure")
+
+        assert isinstance(model.fs.unit.retained_liquid_outlet, Port)
+        assert len(model.fs.unit.retained_liquid_outlet.vars) == 4
+        assert hasattr(model.fs.unit.retained_liquid_outlet, "flow_vol")
+        assert hasattr(model.fs.unit.retained_liquid_outlet, "conc_mol_comp")
+        assert hasattr(model.fs.unit.retained_liquid_outlet, "temperature")
+        assert hasattr(model.fs.unit.retained_liquid_outlet, "pressure")
+
+        assert isinstance(model.fs.unit.recovered_liquid_outlet, Port)
+        assert len(model.fs.unit.recovered_liquid_outlet.vars) == 4
+        assert hasattr(model.fs.unit.recovered_liquid_outlet, "flow_vol")
+        assert hasattr(model.fs.unit.recovered_liquid_outlet, "conc_mol_comp")
+        assert hasattr(model.fs.unit.recovered_liquid_outlet, "temperature")
+        assert hasattr(model.fs.unit.recovered_liquid_outlet, "pressure")
+
+        assert isinstance(model.fs.unit.split, SeparatorData)
+        assert isinstance(model.fs.unit.liquid_recovery, Var)
+
+        print(number_variables(model))
+        print(number_total_constraints(model))
+        print(number_unused_variables(model))
+
+        assert number_variables(model) == 34
+        assert number_total_constraints(model) == 17
+        # These are the solid properties, as they do not appear in constraints
+        assert number_unused_variables(model) == 8
+
+    @pytest.mark.component
+    def test_units(self, model):
+        assert_units_consistent(model)
+
+    @pytest.mark.unit
+    def test_dof(self, model):
+        assert degrees_of_freedom(model) == 0
+
+    @pytest.mark.component
+    def test_structural_issues(self, model):
+        dt = DiagnosticsToolbox(model)
```

Comment:
Is this going to be standard procedure for tests now? I like it; I'm just asking.